packages/map-template/src/hooks/useMapClick.js

```diff
@@ -0,0 +1,53 @@
+import { useEffect, useState } from 'react';
+import mapTypeState from '../atoms/mapTypeState';
+import { useRecoilValue } from 'recoil';
+
+const MAP_PROVIDERS = {
+    MAPBOX: 'mapbox',
+    GOOGLE: 'google'
+}
+
+export default function useMapClick(mapsIndoorsInstance) {
```

Comment:
Write a JSDoc block for this function. May the AI be with you ⚔️ 

Also, consider "sharpening" the name of the hook: The current one may sound like it handles any kind of map clicks.